### PR TITLE
Reduce memory allocation on Geometry.fromBufferGeometry

### PR DESCRIPTION
**Pull request by erasta**
On `Geometry.fromBufferGeometry` various temporary vectors are allocated for organizing the data before populating `Geometry`'s members. The vectors are `tempNormals`, `tempUVs`, and `tempUVs2`.

This change removes those vectors and populates the members directly, using `Vector3.fromArray`

This relates to issue #14394, even though i have no connection to him.

To test this, i built the lib and ran https://threejs.org/examples/webgl_physics_volume.html locally, since it uses `fromBufferGeometry`.

### CODE REVIEW

File: src/core/Geometry.js [hunk 1]
```diff
@@ -228,58 +228,49 @@ Geometry.prototype = Object.assign( Object.create( EventDispatcher.prototype ),
...
 			}
 
 		}
 
 		function addFace( a, b, c, materialIndex ) {
 
-			var vertexNormals = normals !== undefined ? [ tempNormals[ a ].clone(), tempNormals[ b ].clone(), tempNormals[ c ].clone() ] : [];
 			var vertexColors = colors !== undefined ? [ scope.colors[ a ].clone(), scope.colors[ b ].clone(), scope.colors[ c ].clone() ] : [];
 
+			var vertexNormals = normals === undefined ? [] : [
```
Comment (Mugen87) [inline src/core/Geometry.js]: For better readability:
```js
var vertexNormals = ( normals === undefined ) ? [] : [
```

**Review by Mugen87 — CHANGES_REQUESTED** [review]

File: src/core/Geometry.js (hunk 1 above)
Comment (erasta) [reply src/core/Geometry.js]: @Mugen87 Great, Done.
Naturally, this depends on a way of looking.
Maybe we should add this to the guidelines?
Thanks.

Comment (Mugen87) [reply src/core/Geometry.js]: It would be great if you can add the brackets.

Comment (erasta) [reply src/core/Geometry.js]: @Mugen87 Am unsure, did you mean the parentheses around the predicate?
like this: `var vertexNormals = (normals !== undefined) ? [...] : [];`

Comment (Mugen87) [reply src/core/Geometry.js]: I mean just `( normals === undefined )`.

Comment (erasta) [reply src/core/Geometry.js]: No problem :-)

**Review by Mugen87 — APPROVED** [review]

**Comment by erasta** [comment]
Hi @Mugen87, is there anything more that i can do to get this merged? thanks

**Comment by Mugen87** [comment]
No, it's all fine. @mrdoob still needs to review the PR.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**